Source/santad/DataLayer/SNTRuleTable.mm

```diff
@@ -589,4 +591,25 @@ - (BOOL)fillError:(NSError **)error code:(SNTRuleTableError)code message:(NSStri
   return rules;
 }
 
+- (NSString *)hashOfHashes {
+  __block CC_SHA256_CTX sha;
+  CC_SHA256_Init(&sha);
+
+  [self inDatabase:^(FMDatabase *db) {
+    FMResultSet *rs =
+      [db executeQuery:@"SELECT * FROM rules WHERE type!=?", @(SNTRuleStateAllowTransitive)];
+    while ([rs next]) {
+      SNTRule *r = [self ruleFromResultSet:rs];
+      NSString *digest = r.digest;
+      CC_SHA256_Update(&sha, digest.UTF8String, (CC_LONG)digest.length);
```

Comment:
If unicode characters are present, `digest.UTF8String` could be less than `digest.length`. Since these are hex digests that should never happen, but for robustness / completeness `NSStringToUTF8StringView` or `-lengthOfBytesUsingEncoding:` should be used.